content/how-to/gql/call-the-graphql-api.md

```diff
@@ -10,40 +10,433 @@ menu:
 
-## Procedure
+## Call syntax
 
-<!-- if procedure is very long, consider using h3s -->
+The following sections show you the essential features to make a query.
 
-1.
-1.
+### Authentication
 
-<!--Define what success looks like -->
+To authenticate your requests, pass a bearer token as an `Authorization` header.
+Be sure to preface the value with the word `Bearer `:
 
-<!-- 
-### Error recovery
-Optional. Define common failure modes and how to recover
--->
 
-<!--
-## Next steps
-Optional. Summarize what was done. Give an example of what success looks like.
+{{< figure
+alt="Example of how it looks in Apollo explorer"
+src="/images/screenshot-rhize-apollo-auth-headers.png"
+width="50%"
+>}}
+
+For an overview of how Rhize handles token exchange, read [About OpenID connect]({{< relref "/deploy/about-openidconnect" >}}).
+
+### Request body
+
+By default, all GraphQL operations have the following structure:
+1. Define the operation type (one of, `query`, `mutation`, or `subscription`).
+1. Name the query anything you want. This example builds a `query` called `myCustomName`:
+    ```graphql
+    query myCustomName {
+        #operations will go here
+      }
+
+    ```
+1. In curly brackets, define the _operation_ you want to query.
+1. In parenthesis next to the operation, add the _arguments_ to pass. This example uses the `getEquipment` operation, and its arguments specify which item of equipment to get.
+
+    ```graphql
+    query myCustomName {
+      getEquipment(id: "Kitchen_mixer_b_01") {
+        #Fields go here
+        }
+      }
+    ```
+
+1. Within the operation, define the fields you want to return. This example queries for the equipment ID and the person who created the entity.
+
+    ```graphql
+    query myCustomName {
+      getEquipment(id: "Kitchen_mixer_b_01") {
+        id
+        _createdBy
+      }
+    ```
+
+   As you might expect, the request returns only these fields for the equipment named `Kitchen_mixer_b_01`.
+
+   ```json
+   {
+      "data": {
+        "getEquipment": {
+          "id": "Kitchen_mixer_b_01",
+          "_createdBy": "john_snow"
+        }
+      }
+    }
+    ```
+
+### Request exactly the data you want
+
+A major benefit of GraphQL is that you can modify queries to return only the fields you want.
+Unlike calls to REST APIs, where the server-side code defines what a response looks like, GraphQL calls instruct the server to return only what is specified.
+Furthermore, you can query diverse sets of data in one call, so you can get exactly the entities you want without calling multiple endpoints, as you would in REST, or composing queries with complex recursive joins, as you would in SQL.
+Besides precision, this also brings performance benefits to minimize network calls and their payloads.
+
+For example, you modify the previous request to return information about the equipments versions, including their `description`, unique IDs, and the currently active version:
+
+{{< tabs >}}
```

Comment:
I like how the tabs separate the query from the variables, i just don't like that i actually have to click the tab to see the data. Is there another UI pattern that we can use that separates the query from the variables that show it all on one page? Columns or something like that?

It just breaks the flow a little bit from reading to actually having to interact with the page just to continue reading. Some uses might not even click on it and ignore the tab and be confused.

Reply:
> Is there another UI pattern that we can use that separates the query from the variables that show it all on one page? Columns or something like that?

This is definitely something that we could implement.  For example, how Graphql.org does. Though personally, I'd rather only have two columns, then put the result in a separate tab.

![image](https://github.com/libremfg/libremfg.github.io/assets/47385188/84899287-8754-4a5b-bada-2ff141aa84f2)


However, this is definitely not something I could do by week's end. 

>It just breaks the flow a little bit from reading to actually having to interact with the page just to continue reading. Some uses might not even click on it and ignore the tab and be confused.

I did worry about this, too. Until we have a better solution, the alternative is just put both in one code block.

Here's the longest example query. It's not the end of the world, but the page is a little long.

![image](https://github.com/libremfg/libremfg.github.io/assets/47385188/0869c892-9613-4814-8af3-5f26e9c32145)